Sources/MongoSwift/BSON/Document+Collection.swift

```diff
@@ -0,0 +1,42 @@
+import Foundation
+
+/// An extension of `Document` to make it conform to the `Collection` protocol.
+/// This gives guarantees on non-destructive iteration, and offers an indexed
+/// ordering to the key value pairs in the document.
+extension Document: Collection {
+    /// Returns the start index of the Document.
+    public var startIndex: Int {
+        return 0
+    }
+
+    /// Returns the end index of the Document.
+    public var endIndex: Int {
+        precondition(self.count > 0)
```

Comment:
this seems kind of weird - are you just not supposed to call `endIndex` until you've put something in the doc?

Reply:
Done.